website/src/pages/api/admin/update_user.ts

```diff
@@ -0,0 +1,29 @@
+import { getToken } from "next-auth/jwt";
+import prisma from "src/lib/prismadb";
+
+/**
+ * Update's the user's data in the database.  Accessible only to admins.
+ */
+const handler = async (req, res) => {
+  const token = await getToken({ req });
+
+  // Return nothing if the user isn't registered or if the user isn't an admin.
+  if (!token || token.role !== "admin") {
+    res.status(403).end();
+    return;
+  }
```

Comment:
could we extract these to a middleware?

maybe something like:


```ts
const handler = withAdmin(async (token, req, res) => {
...
});
```
or maybe
```ts
```ts
const handler = withRole("admin", async (token, req, res) => {
...
});
```

Reply:
Done.  I looked at using the actual `middleware.ts` but it looks a bit too messy.  A wrapper is a lot cleaner.  makes me miss RedwoodJS where they built this into the graphQL handler.